src/core/geometry/qgscurvepolygon.cpp

```diff
@@ -1478,3 +1542,38 @@ int QgsCurvePolygon::compareToSameClass( const QgsAbstractGeometry *other ) cons
 
   return 0;
 }
+
+bool QgsCurvePolygon::hasPlane3D( QgsPoint &ptA, QgsPoint &ptB, QgsPoint &ptC, double epsilon ) const
+{
+  if ( !mExteriorRing || !exteriorRing()->is3D() )
+  {
+    return false;
+  }
+
+  /*
+   * look for 3 non collinear points on the exterior ring
```

Comment:
I think this logic is broken -- what would happen if the first 3 points are in the same plane, but then the next point isn't? With the current logic you'll happily return true.

What about rings with a different plane?

Sure, these aren't valid 2.5d geometries, but the function naming and docs misleading suggest that this can be used to test that the polygon IS in a single plane, and it doesn't do that right now...

Reply:
This is the intended behavior. I have updated the documentation of `has3DPlane` to explain the behavior.